services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDocumentBuilder.java

```diff
@@ -126,4 +131,54 @@ public MongoDbDocumentBuilder withAuthId(final String authId) {
         document.put(FIELD_CREDENTIALS_AUTH_ID_KEY, authId);
         return this;
     }
+
+
+    /**
+     * Sets the json object with the given device filters.
+     *
+     * @param filters The device filters list.
+     * @return a reference to this for fluent use.
+     */
+    public MongoDbDocumentBuilder withDeviceFilters(final List<Filter> filters) {
+
+        filters.forEach(filter -> {
+            // TODO: To implement when filter values contain patterns such as * or %
+            if (FIELD_ID.equals(filter.getField())) {
+                document.put(RegistryManagementConstants.FIELD_PAYLOAD_DEVICE_ID, filter.getValue());
+            } else {
+                document.put(MongoDbDeviceRegistryUtils.FIELD_DEVICE + filter.getField().toString().replace("/", "."),
```

Comment:
does this also cover indexes into arrays, i.e. something like `/myJsonArray/1` to address the second element of a JSON array?

Reply:
yes, it covers arrays too. For example ` /myJsonArray/1` is translated to `device.myJsonArray.1` as required by the _MongoDB_ search operation.